### ISSUE TITLE
Add option to execute model from explorer

### ISSUE THREAD

**Issue by odjuricicTT** [tenstorrent/tt-mlir#1020]
This might require changes in ttrt in order to be able to use ttrt.API.perf.

**Comment by odjuricicTT** [comment]
@vprajapati-tt What do you think about adding `--output-file` (or similar) to ttrt perf, in other to not have to parse the output of a subprocess.

**Comment by vprajapati-tt** [comment]
Hey! I just saw this, a while back I added the `--artifact-dir` option when running perf. TTRT itself outputs the log to a file (if specified) and also `stdin`. The issue with Perf is that TTRT ends up using `signal` to look for `SIGINT` and `SIGTERM` from Tracy. Since `model-explorer` is run in a thread (or process), this isn't well received by python and it either crashes or doesn't detect these signals. I played around with this for a while and circled back to using a `subprocess` since that was the simplest method to get a prototype working. 

I'd be interested to revisit this as there's probably something I'm missing that can solve this problem. Let me know if there's anything else I can clarify.
